### DISCUSSION TITLE
Delete Object after querying for it

### DISCUSSION THREAD

**Discussion by DavidJakszta** [prisma/prisma#19919] [Q&A] [answered]
so lets say i query for a user and then put it into a variable called dbUser. Is there some way to delete that user faster for example with something like `dbUser.delete()` or something. Otherwise the only choice is to do another delete query right?

```js
        const dbUser = await db.user.findFirst({
            where: {
                id: sessionUser.id
            }
        });
```

**Comment by nurul3101** [comment, marked as answer]
Hello @wtfmynamefits 👋 

There's no mechanism to delete a record from the database directly through the object instance returned from a query (like `dbUser.delete()` in your example). Instead, you would typically need to issue a [delete](https://www.prisma.io/docs/concepts/components/prisma-client/crud#delete) query.

Here's how you can delete the user record using Prisma after retrieving it:

```js
const dbUser = await db.user.findFirst({
    where: {
        id: sessionUser.id
    }
});

if (dbUser) {
    await db.user.delete({
        where: {
            id: dbUser.id
        }
    });
}
```

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
